source/MRMesh/MRPlaneObject.cpp

```diff
@@ -0,0 +1,139 @@
+#include "MRPlaneObject.h"
+#include "MRMesh.h"
+#include "MRMeshBuilder.h"
+#include "MRBestFit.h"
+#include "MRObjectFactory.h"
+#include "MRPch/MRJson.h"
+#include "MRMatrix3.h"
+
+namespace MR
+{
+
+MR_ADD_CLASS_FACTORY( PlaneObject )
+
+Vector3f PlaneObject::getNormal() const
+{
+    return ( xf().A * Vector3f::plusZ() ).normalized();
+}
+
+Vector3f PlaneObject::getCenter() const
+{
+    return xf().b;
+}
+
+void PlaneObject::setNormal( const Vector3f& normal )
+{
+    auto currentXf = xf();
+    currentXf.A = Matrix3f::rotation( Vector3f::plusZ(), normal ) * Matrix3f::scale( currentXf.A.toScale() );
+    setXf( currentXf );
+}
+
+void PlaneObject::setCenter( const Vector3f& center )
+{
+    auto currentXf = xf();
+    currentXf.b = center;
+    setXf( currentXf );
+}
+
+void PlaneObject::setSize( float size )
+{
+    auto currentXf = xf();
+    currentXf.A = Matrix3f::rotationFromEuler( currentXf.A.toEulerAngles() ) * Matrix3f::scale( Vector3f::diagonal( size ) );
+    setXf( currentXf );
+}
+
+PlaneObject::PlaneObject()
+{
+    constructMesh_();
+}
+
+PlaneObject::PlaneObject( const std::vector<Vector3f>& pointsToApprox )
+{
+    constructMesh_();
+
+    PointAccumulator pa;
+    Box3f box;
+    for ( const auto& p : pointsToApprox )
+    {
+        pa.addPoint( p );
+        box.include( p );
+    }
+
+    Plane3f plane = pa.getBestPlanef();
+    Vector3f normal = plane.n.normalized();
+    if ( plane.d < 0 )
+        normal *= -1.f;
+
+    setNormal( normal );
+    setCenter( box.center() );
```

Comment:
we can find center by `plane.d*plane.n` no need to accum box

Reply:
center as bbox center need for right visualization